File: 251001/Levitskij/DistComp/module-discussion/src/main/java/by/kopvzakone/distcomp/entities/Reaction.java
package by.kopvzakone.distcomp.entities;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Reference;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Indexed;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table
public class Reaction {
    @PrimaryKey
    private ReactionKey id;
    @Size(min = 2, max = 32)
    String content;
    @Indexed
    private long tweetId;

}

File: 251001/Vasilenko/discussion/src/main/kotlin/org/ex/discussion/model/Reaction.kt
package org.ex.discussion.model

import lombok.Getter
import lombok.Setter
import org.springframework.data.cassandra.core.mapping.Column
import org.springframework.data.cassandra.core.mapping.PrimaryKey
import org.springframework.data.cassandra.core.mapping.Table

@Setter
@Getter
@Table("reaction")
data class Reaction (

    @PrimaryKey
    @Column("id")
    val id: Long,

    @Column("country")
    val country: String,

    @Column("tweetId")
    var tweetId: Long,

    @Column("content")
    var content: String
)